packages/sdk/src/types/TransactionExpiry.ts

```diff
@@ -28,11 +24,32 @@ class TransactionExpiry {
         public readonly expiryEpochSeconds: bigint
     ) {}
 
-    public toJSON(): number {
-        return Number(this.expiryEpochSeconds);
+    /**
+     * Get a string representation of the transaction expiry date in seconds since the Unix epoch.
+     * @returns {string} The string representation.
+     */
+    public toString(): string {
+        return this.expiryEpochSeconds.toString();
+    }
+
+    /**
+     * Get a JSON-serializable representation of the transaction expiry date.
+     * @returns {string} The JSON-serializable representation.
+     */
+    public toJSON(): string {
+        return this.expiryEpochSeconds.toString();
```

Comment:
So this used to be serialized as a `number`, but is now a `string`? I don't know if we should do this, as it might have unintended consequences (the output of `JSON.stringify` will change without warning).

I would (again) prefer that we change this to return a `bigint`, which at least brings the error to the user at the point of serializing the type.

@limemloh @lassemoldrup what do you think we should do here?

Reply:
The problem is that converting to a number can fail if the internal `bigint` is too large. We discussed this and decided that converting to a `number` should be considered a bug, thus it was changed.

As to whether `string` or `bigint` is the right type, I think that since this is a one-off case, having a type that you can call `JSON.stringify` on is more important than any compatibility concerns. If you want the `bigint`, it is also easily accessible.

Reply:
What I mean is that we have existing infrastructure that might depend on this returning a `number` type since it has been declared (as opposed to the many types that had no `toJSON` method).

Taking the pragmatic standpoint, the current (faulty) implementation will not fail as it will be a very long time until `Number.MAX_SAFE_INTEGER` is hit with seconds from unix epoch.

```
new Date(Number.MAX_SAFE_INTEGER) // Invalid date
new Date(Number.MAX_SAFE_INTEGER / 10) // Mon Sep 05 30512 05:17:54 GMT+0200 (Central European Summer Time)
```

I would rather keep the faulty implementation until we can introduce this as a breaking change if we want to 🤷

Reply:
After discussion with @limemloh, we decided to keep `number` and throw an error if the value is not a safe integer.